CRM/Core/DAO.php

```diff
@@ -2268,6 +2268,54 @@ public static function getReferencesToTable($tableName) {
     return $refsFound;
   }
 
+  /**
+   * Get all references to contact table.
+   *
+   * This includes core tables, custom group tables, tables added by the merge
+   * hook and  the entity_tag table.
+   *
+   * Refer to CRM-17454 for information on the danger of querying the information
+   * schema to derive this.
+   */
+  public static function getReferencesToContactTable() {
+    if (isset(\Civi::$statics[__CLASS__]) && isset(\Civi::$statics[__CLASS__]['contact_references'])) {
```

Comment:
@totten so I'm curious about converting this to a cache call

In theory the things that would change the list of tables here are
1) a new custom table being created
2) a change to core schema
3) a change to hook-declared entity types. 

3 breaks down into 
a) new extension installed
b) new extension uninstalled
c) extension upgrade WITH upgrade extension script run
d) extension upgrade without any script run
e) main civicrm upgrade script runs and hooks are called in upgrade mode - causing the list of cached extension functions to be built without extension hooks being called and hence to be missing some

Reply:
(note I think this PR is mergeable regardless of my tangent)